src/test/java/teammates/test/cases/datatransfer/AdminEmailAttributesTest.java

```diff
@@ -28,8 +28,77 @@
     private String subject = "subject of email";
     private Text content = new Text("valid email content");
     private Date date = new Date();
-    private AdminEmailAttributes adminEmailAttributes = new AdminEmailAttributes(
-            subject, addressReceiverListString, groupReceiverListFileKey, content, date);
+    private AdminEmailAttributes adminEmailAttributes = AdminEmailAttributes
+            .builder(subject, addressReceiverListString, groupReceiverListFileKey, content)
+            .withSendDate(date)
+            .build();
+
+    @Test
+    public void testBuilderWithDefaultValues() {
+        ______TS("valid admin email");
+
+        AdminEmailAttributes attributesWithDefaultValues = AdminEmailAttributes
+                .builder(subject, addressReceiverListString, groupReceiverListFileKey, content)
+                .build();
+
+        ______TS("success: default values for optional params");
+
+        assertEquals(Const.ParamsNames.ADMIN_EMAIL_ID, attributesWithDefaultValues.getEmailId());
+        assertFalse("Default false for isInTrashBin", attributesWithDefaultValues.isInTrashBin);
+        assertEquals(Const.TIME_REPRESENTS_DEFAULT_TIMESTAMP, attributesWithDefaultValues.getCreateDate());
+        assertTrue("Valid input", attributesWithDefaultValues.isValid());
+    }
+
+    @Test
+    public void testBuilderWithNullOptionalArguments() {
+        AdminEmailAttributes attributesWithNullOptionalArguments = AdminEmailAttributes
+                .builder(subject, addressReceiverListString, groupReceiverListFileKey, content)
+                .withSendDate(null)
+                .withCreateDate(null)
+                .withEmailId(null)
+                .withIsInTrashBin(null)
+                .build();
+
+        ______TS("success: default values for optional params");
+
+        assertEquals(Const.ParamsNames.ADMIN_EMAIL_ID, attributesWithNullOptionalArguments.getEmailId());
+        assertFalse("Default false for isInTrashBin", attributesWithNullOptionalArguments.isInTrashBin);
+        assertEquals(Const.TIME_REPRESENTS_DEFAULT_TIMESTAMP, attributesWithNullOptionalArguments.getCreateDate());
+        assertEquals(null, attributesWithNullOptionalArguments.getSendDate());
+
+        ______TS("valid admin email");
+
+        assertTrue("Valid input", attributesWithNullOptionalArguments.isValid());
+    }
+
+    @Test
+    public void testBuilderWithNullRequiredArguments() {
+        try {
+            AdminEmailAttributes attributesWithNullRequiredArguments = AdminEmailAttributes
+                        .builder(null, null, null, null)
+                        .build();
+
+            attributesWithNullRequiredArguments.isValid();
+            signalFailureToDetectException(" - AssertionError");
+        } catch (AssertionError ae) {
+            assertEquals("Non-null value expected for required AdminEmailAttributes", ae.getMessage());
+        }
```

Comment:
No need here for try-catch block. You can add something like that
`@Test(expectedExceptions = AssertionError.class, expectedExceptionsMessageRegExp = "Non-null value expected for required AdminEmailAttributes")` annotation on this method